src/lib.rs

```diff
@@ -125,6 +135,74 @@ pub trait StringSlice {
     /// [`Option`]: https://doc.rust-lang.org/std/option/enum.Option.html
     /// [`None`]: https://doc.rust-lang.org/std/option/enum.Option.html#variant.None
     fn try_substring(&self, begin: usize, end: usize) -> Option<&str>;
+
+    #[cfg(feature = "rayon")]
+    /// Returns a string slice for the given range of characters
+    ///
+    /// This method will panic if the range is invalid,
+    /// for example if the beginning is greater than the end.
+    ///
+    /// Runs in parallel
```

Comment:
```suggestion
    /// Uses [`rayon`] to find character positions in parallel
    ///
    /// [`rayon`]: https://github.com/rayon-rs/rayon
```

Just to make it clear what is being parallelised